contracts/schemes/GenericSchemeMultiCall.sol

```diff
@@ -0,0 +1,197 @@
+
+    IntVoteInterface public votingMachine;
+    bytes32 public voteParams;
+    mapping(address=>bool) internal contractWhitelist;
+    address[] public whitelistedContracts;
+    Avatar public avatar;
+
+    /**
+     * @dev initialize
+     * @param _avatar the avatar to mint reputation from
+     * @param _votingMachine the voting machines address to
+     * @param _voteParams voting machine parameters.
+     * @param _contractWhitelist the contracts the scheme is allowed to interact with
+     * 
+     */
+    function initialize(
+        Avatar _avatar,
+        IntVoteInterface _votingMachine,
+        bytes32 _voteParams,
+        address[] calldata _contractWhitelist
+    )
+    external
+    {
+        require(avatar == Avatar(0), "can be called only one time");
+        require(_avatar != Avatar(0), "avatar cannot be zero");
+        require(_contractWhitelist.length > 0, "contractWhitelist cannot be empty");
+        avatar = _avatar;
+        votingMachine = _votingMachine;
+        voteParams = _voteParams;
+        for(uint i = 0; i < _contractWhitelist.length; i ++) {
+          contractWhitelist[_contractWhitelist[i]] = true;
+          whitelistedContracts.push(_contractWhitelist[i]);
+        }
+    }
+
+    /**
+    * @dev execution of proposals, can only be called by the voting machine in which the vote is held.
+    * @param _proposalId the ID of the voting in the voting machine
+    * @param _decision a parameter of the voting result, 1 yes and 2 is no.
+    * @return bool success
+    */
+    function executeProposal(bytes32 _proposalId, int256 _decision)
+    external
+    onlyVotingMachine(_proposalId)
+    returns(bool) {
+        MultiCallProposal storage proposal = proposals[_proposalId];
+        require(proposal.exist, "must be a live proposal");
+        require(proposal.passed == false, "cannot execute twice");
+
+        if (_decision == 1) {
+            proposal.passed = true;
+            execute(_proposalId);
+        } else {
+            delete proposals[_proposalId];
+            emit ProposalDeleted(address(avatar), _proposalId);
+        }
+
+        emit ProposalExecutedByVotingMachine(address(avatar), _proposalId, _decision);
+        return true;
+    }
+
+    /**
+    * @dev execution of proposals after it has been decided by the voting machine
+    * @param _proposalId the ID of the voting in the voting machine
+    */
+    function execute(bytes32 _proposalId) public {
+        MultiCallProposal storage proposal = proposals[_proposalId];
+        require(proposal.exist, "must be a live proposal");
+        require(proposal.passed, "proposal must passed by voting machine");
+        proposal.exist = false;
+        bytes memory genericCallReturnValue;
+        bool success;
+        Controller controller = Controller(avatar.owner());
+
+        for (uint i = 0; i < proposal.contractsToCall.length; i ++) {
+          if (proposal.contractsToCall[i] == address(controller)) {
```

Comment:
why is this condition check needed? The controller can just called the token approval to the avatar with the encoded data, right?

Reply:
see this convo: https://github.com/daostack/arc/pull/784#discussion_r480865631

Reply:
Just commented it, I think there is a misunderstanding there because the avatar will call the ERC20 token, no the controller. 

But if you dont check that you have to list every ERC20 token that you want to call, I guess calling the external token transfer in the controller allows to call any token, it is like a "whitelisted" function

Reply:
@nicoelzer does any how the plan is to allow the avatar to transfer any erc20 token its own?. this might be the same as approval ... if this is the case we might do not need this special approval staff. as @AugustoL mention. (though will need to whitelist that somehow).
If the case is that only approval is needed for any erc20 so we can avoid white listing the erc20 and have the solution as it is now .